sdk/servicebus/azure-servicebus/api_review.py

```diff
@@ -0,0 +1,341 @@
+# Two-Clients Approach
+
+## APIs:
+class SeviceBusSenderClient:
+    def __init__(
+        self,
+        fully_qualified_namespace : str,
+        entity_name : str,
+        credential : TokenCredential,
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        transport_type: TransportType = None,
+        retry_total : int = 3,
+        retry_backoff_factor : float = 30,
+        retry_backoff_maximum : int = 120  # cur_retry_backoff = retry_backoff_factor * (2^cur_retry_time)
+        connection : ServiceBusConnection = None
+    ) -> None:
+    @classmethod
+    def from_queue(
+        cls,
+        fully_qualified_namepsace : str,
+        queue_name : str,
+        credential : TokenCredential,
+        **kwargs
+    ) -> SeviceBusSenderClient:
+    @classmethod
+    def from_topic(
+        cls,
+        fully_qualified_namespace : str,
+        topic_name : str,
+        credential : TokenCredential,
+        **kwargs
+    ) -> SeviceBusSenderClient:
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str : str,
+        entity_name : str = None,
+        **kwargs
+    ) -> SeviceBusSenderClient:
+
+    def __enter__(self):
+    def __exit__(self):
+
+    def close(self) -> None:
+
+    def get_properties(self) -> Dict[str, Any]:
+
+    def list_session_ids(self) -> List[str]:
+
+    def create_batch(
+        self,
+        session_id : str = None,
+        max_size_in_bytes : int = None
+    ) -> BatchMessage:
+
+    def send(
+        self,
+        message : Union[Message, BatchMessage],
+        session_id : str = None,
+        message_timeout : float = None
+    ) -> None:
+    def schedule(
+        self,
+        message : Union[Message, BatchMessage],
+        schedule_time : datetime,
+        session_id : str = None
+    ) -> List[int]:
+
+    def cancel_scheduled_messages(self, sequence_number : Union[int, List[int]]) -> None:
+
+
+class ServiceBusReceiverClient:
+    def __init__(
+        self,
+        fully_qualified_namespace : str,
+        entity_name : str,
+        credential: TokenCredential,
+        subscription_name : str = None,
+        session_id : str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        transport_type: TransportType = None,
+        retry_total : int = 3,
+        retry_backoff_factor: float = 30,
+        retry_backoff_maximum: int = 120  # cur_retry_backoff = retry_backoff_factor * (2^cur_retry_time)
+        connection: Connection = None
+    ) -> None:
```

Comment:
Thoughts on dead letter receiver:

- constructor parameter
- extra method on ReceiverClient for dead letter queue

Reply:
Perhaps just an extra getDeadLetterReceiver() method on your top level client?

Reply:
Closing this as it's not in the scope of preview 1.